internal/test/builder/builders.go

```diff
@@ -427,38 +428,42 @@ func InfrastructureMachineTemplate(namespace, name string) *InfrastructureMachin
 //     "spec.version": "v1.2.3",
 // }.
 func (i *InfrastructureMachineTemplateBuilder) WithSpecFields(fields map[string]interface{}) *InfrastructureMachineTemplateBuilder {
-	i.specFields = fields
+	for field, value := range fields {
+		if err := unstructured.SetNestedField(i.obj.UnstructuredContent(), value, field); err != nil {
+			panic(err)
+		}
 	return i
 }
 
 // Build takes the objects and variables in the  InfrastructureMachineTemplateBuilder and generates an unstructured object.
 func (i *InfrastructureMachineTemplateBuilder) Build() *unstructured.Unstructured {
-	obj := &unstructured.Unstructured{}
-	obj.SetAPIVersion(InfrastructureGroupVersion.String())
-	obj.SetKind(GenericInfrastructureMachineTemplateKind)
-	obj.SetNamespace(i.namespace)
-	obj.SetName(i.name)
-
-	// Initialize the spec.template.spec to make the object valid in reconciliation.
-	setSpecFields(obj, map[string]interface{}{"spec.template.spec": map[string]interface{}{}})
-
-	setSpecFields(obj, i.specFields)
-	return obj
+	if _, ok, _ := unstructured.NestedMap(i.obj.Object, "spec"); !ok {
+		if err := unstructured.SetNestedField(i.obj.Object, map[string]interface{}{}, "spec"); err != nil {
```

Comment:
Could you fix the build errors in this PR and make sure it compiles?

Reply:
@killianmuldoon I tried fixing the errors, but I am not really able to figure out where I am making the mistake. I followed the exact steps from the demo PR, but I am not really getting it.

Reply:
@sayantani11 There's a number of errors here that mean the code won't compile - I think your best resource for figuring out the issues might be to look at https://go.dev/tour/welcome/1 to get a feel for some of the contructs. Otherwise the compiler should print out error messages for these that show what the issue is.

 I'll explain a couple of them in comments, but I think it might help if you spend a bit of time on tour of go and the go playground to help you understand some of the syntax issues here.

Reply:
Ok thanks for this I'll look over it